### PR TITLE
module names are made more consistent with POI page

### PR DESCRIPTION
**Pull request by b0noI**
Less intrusive update after the first attempt got reverted: https://github.com/pytorch/pytorch/pull/83127

fix for: #83363

### CODE REVIEW

File: .github/merge_rules.json [hunk 1]
```diff
@@ -1,6 +1,16 @@
...
+      "patterns": ["*"],
+      "approved_by": ["soumith", "gchanan", "ezyang", "dzhulgakov"],
+      "mandatory_checks_name": [
+         "Facebook CLA Check",
+         "Lint",
+         "pull"
+      ]
+   },
+   {
+      "name": "PyTorch -> ONNX",
```
Comment (malfet) [inline .github/merge_rules.json:13]: Please keep ONNX exporter as name
```suggestion
      "name": "ONNX exporter",
```

File: .github/merge_rules.json [hunk 2]
```diff
@@ -26,14 +36,14 @@
       ]
    },
    {
-      "name": "NVFuser",
+      "name": "NVIDIA",
```
Comment (malfet) [inline .github/merge_rules.json:39]: Please keep the old name in place
```suggestion
      "name": "NVfuser",
```

File: .github/merge_rules.json [hunk 3]
```diff
@@ -163,14 +156,14 @@
...
-      "name": "MPS",
+      "name": "Apple M1/MPS",
       "patterns": [
          "test/test_mps.py",
          "aten/src/ATen/native/native_functions.yaml",
          "aten/src/ATen/mps/**",
          "aten/src/ATen/native/mps/**"
       ],
-      "approved_by": ["kulinseth", "razarmehr"],
+      "approved_by": ["kulinseth", "alband", "malfet"],
```
Comment (malfet) [inline .github/merge_rules.json:166]: Why do you want to remove @razarmehr from this list?

**Review by malfet — CHANGES_REQUESTED** [review]
Please fix the lint errors and also provide some reasoning about renames/regrouping in PR description

IMO removing separate rules for pin updates are wrong, because:
 - They were created to limit files that could be changed by bot (this PR extends it to the entire .github folder(
 - Specific XLA rule is needed to prevent pin updates when XLA jobs are disabled for one reason or another

File: .github/merge_rules.json [hunk 4]
```diff
@@ -42,42 +52,25 @@
...
-      "name": "XLA hash pin update",
-      "patterns": [".github/ci_commit_pins/xla.txt"],
-      "approved_by": ["pytorchbot", "ezyang", "pytorch/pytorch-dev-infra"],
-      "mandatory_checks_name": [
-         "Facebook CLA Check",
-         "Lint",
-         "pull / linux-bionic-py3_7-clang8-xla / build",
-         "pull / linux-bionic-py3_7-clang8-xla / test (xla, 1, 1, linux.2xlarge)"
-      ]
-   },
```
Comment (malfet) [inline .github/merge_rules.json:76]: Please keep separate rule for XLA hash pin update (but probably only for the bot, as XLA tests are sometimes disabled in pull and we don't want updates to happen during that time)

File: .github/merge_rules.json [hunk 5]
```diff
@@ -42,42 +52,25 @@
...
-         ".github/ci_commit_pins/torchdynamo.txt"
+         ".github/**",
+         ".circleci/**",
+         ".jenkins/**",
+         "scripts/**",
+         "tools/**",
+         ".github/ci_commit_pins/*"
       ],
-      "approved_by": ["pytorchbot", "ezyang", "pytorch/pytorch-dev-infra"],
+      "approved_by": ["alband", "dagitses", "pytorchbot", "pytorch/pytorch-dev-infra"],
```
Comment (malfet) [inline .github/merge_rules.json:63]: This change make "pytorchnbot" as an approved reviewer for everythign in CI, but for security/sanity reasons I would prefer bot to be limited to just pinned hashes

File: .github/merge_rules.json [hunk 6]
```diff
@@ -87,7 +80,7 @@
    {
       "name": "Mobile",
       "patterns": ["ios/**", "android/**", "test/mobile/**"],
-      "approved_by": ["linbinyu", "kit1980", "IvanKobzarev", "dreiss"],
+      "approved_by": ["linbinyu", "xta0", "IvanKobzarev", "dreiss", "raziel"],
```
Comment (malfet) [inline .github/merge_rules.json:83]: Not sure why you want to remove kit1980 nor add xta0 here
```suggestion
      "approved_by": ["linbinyu", "IvanKobzarev", "dreiss", "raziel"],
```

File: .github/merge_rules.json [hunk 7]
```diff
@@ -210,7 +203,19 @@
...
+      "approved_by": [
+         "mrshenli",
+         "pritamdamania87",
+         "zhaojuanmao",
+         "rohan-varma",
+         "wanchaol",
+         "fduwjj",
+         "H-Huang",
+         "d4l3k",
+         "kiukchung",
```
Comment (malfet) [inline .github/merge_rules.json:215]:
```suggestion
```

File: .github/merge_rules.json (hunk 4 above)
Comment (b0noI) [reply .github/merge_rules.json:76]: done

File: .github/merge_rules.json (hunk 5 above)
Comment (b0noI) [reply .github/merge_rules.json:63]: done

File: .github/merge_rules.json (hunk 4 above)
Comment (b0noI) [reply .github/merge_rules.json:76]: I created standalone rule only for the bot with the access to the folder with hash pin files

File: .github/merge_rules.json (hunk 3 above)
Comment (b0noI) [reply .github/merge_rules.json]: added him back in the latest version

**Review by malfet — APPROVED** [review]

File: .github/merge_rules.json [hunk 8]
```diff
@@ -42,42 +52,37 @@
...
       "name": "OSS CI",
-      "patterns": [".github/**", ".circleci/**", ".jenkins/**", "scripts/**", "tools/**"],
-      "approved_by": ["ezyang", "pytorch/pytorch-dev-infra"],
+      "patterns": [
+         ".github/**",
+         ".circleci/**",
+         ".jenkins/**",
+         "scripts/**",
+         "tools/**",
+         ".github/ci_commit_pins/*"
```
[the remark this answers is not in the document]
Comment (b0noI) [reply .github/merge_rules.json]: fixed

File: .github/merge_rules.json [hunk 9]
```diff
@@ -42,42 +52,37 @@
...
-      "mandatory_checks_name": [
-         "Facebook CLA Check",
-         "Lint",
-         "pull / linux-bionic-py3_7-clang8-xla / build",
-         "pull / linux-bionic-py3_7-clang8-xla / test (xla, 1, 1, linux.2xlarge)"
-      ]
-   },
-   {
-      "name": "Documentation",
+      "name": "Docs / Tutorials",
```
[the remark this answers is not in the document]
Comment (b0noI) [reply .github/merge_rules.json:94]: fixed

File: .github/merge_rules.json [hunk 10]
```diff
@@ -108,7 +113,7 @@
          "tools/autograd/templates/python_linalg_functions.cpp",
          "test/test_linalg.py"
       ],
-      "approved_by": ["nikitaved", "mruberry", "pearu", "Lezcano", "IvanYashchuk"],
+      "approved_by": ["mruberry", "Lezcano", "IvanYashchuk"],
```
[the remark this answers is not in the document]
Comment (b0noI) [reply .github/merge_rules.json:127]: they are not in https://github.com/pytorch/pytorch/blob/master/docs/source/community/persons_of_interest.rst#linear-algebra-torchlinalg

File: .github/merge_rules.json [hunk 11]
```diff
@@ -210,7 +215,18 @@
...
+         "mrshenli",
+         "pritamdamania87",
+         "zhaojuanmao",
+         "rohan-varma",
+         "wanchaol",
+         "fduwjj",
+         "H-Huang",
+         "d4l3k",
+         "aazzolini",
+         "kwen2501"
```
[the remark this answers is not in the document]
Comment (b0noI) [reply .github/merge_rules.json:239]: issue created: https://github.com/pytorch/pytorch/issues/83440 (for single source of truth)

[the remark this answers is not in the document]
Comment (b0noI) [reply .github/merge_rules.json:239]: issue created to find way to add comments to merge_rules.json

[the remark this answers is not in the document]
Comment (b0noI) [reply .github/merge_rules.json:239]: https://github.com/pytorch/pytorch/issues/83441

File: .github/merge_rules.json [hunk 12]
```diff
@@ -42,42 +52,37 @@
...
-      "approved_by": ["ezyang", "pytorch/pytorch-dev-infra"],
+      "patterns": [
+         ".github/**",
+         ".circleci/**",
+         ".jenkins/**",
+         "scripts/**",
+         "tools/**",
+         ".github/ci_commit_pins/*"
+      ],
+      "approved_by": ["alband", "dagitses", "pytorch/pytorch-dev-infra"],
```
[the remark this answers is not in the document]
Comment (b0noI) [reply .github/merge_rules.json:62]: yes, I will do this in the next PR, do not want to add one more change to this one

**Pull request closed without merging**